pandas/core/generic.py

```diff
@@ -9226,7 +9226,7 @@ def _tz_convert(ax, tz):
             ax = _tz_convert(ax, tz)
 
         result = self._constructor(self._data, copy=copy)
-        result.set_axis(ax, axis=axis, inplace=True)
```

Comment:
this should really have the inplace arg rather than copy as that is the meaning here.

Reply:
not sure if we should change this api though as don’t really love inplace generally